controllers/clustergroupupgrade_controller.go

```diff
@@ -973,34 +1141,148 @@ func (r *ClusterGroupUpgradeReconciler) reconcileResources(ctx context.Context,
 	return nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) buildRemediationPlan(ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+func (r *ClusterGroupUpgradeReconciler) isClusterNonCompliantWithPolicy(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string, policy *unstructured.Unstructured) bool {
+	// Get the status part of the policy.
+	statusObject := policy.Object["status"].(map[string]interface{})
+	// Get the part of the status that holds the clusters' compliance.
+	statusCompliance := statusObject["status"]
+
+	// If all the clusters are compliant, return nil.
+	if statusCompliance == nil {
```

Comment:
Is a nil status really equivalent to the policy being compliant?

Reply:
To be honest, I am not 100% sure. What I relied upon was the status.status of this policy below which is nil if no clusters are NonCompliant with it (again with the reverse logic here, sorry).
```
status:
  compliant: NonCompliant
  placement:
  - placementBinding: binding-policy1
    placementRule: placement-policy1
  status:
  - clustername: spoke1
    clusternamespace: spoke1
    compliant: NonCompliant
  - clustername: spoke6
    clusternamespace: spoke6
    compliant: NonCompliant
```

I suppose that instead I could check for status.compliant, but my worry is that if somehow there is no placementRule/placementBinding for the policy, then we don't have access to that field either, that's why I chose this option.

Reply:
I think we need to verify with the ACM team how/when the status gets updated. For example, when I first bind the cluster to the policy does it immediately get put into the NonCompliant status, or is there a phase where it is checking the cluster and the cluster is not found in either? Depending on their answer we may need to switch the test to be "positive compliance" rather than not-non-compliant.

Reply:
Still waiting on answer from ACM. Meanwhile, I think it's safe to assume that if we can't access the status part of a policy that includes the clusters (for whatever reason), it's safer to assume the cluster is not compliant actually. That way, even if the cluster is already compliant with the policy, it won't impact the end result.